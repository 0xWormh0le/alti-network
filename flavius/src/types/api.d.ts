type QueryParamValue = string | number | string[] | number[] | null | undefined | boolean
type QueryParams = { [k: string]: QueryParamValue }

type ApiClientReturn<TState extends any, TCallerArgs extends any> = [
  Maybe<TState>,
  Maybe<{ error: string }>,
  boolean,
  ApiCaller<TCallerArgs>
]

type ApiErrorCallback = (err: any) => void
type ApiSuccessCallback = (response: any) => void

interface ApiClientInit {
  defaultPageSize?: number
  pageCountCacheTtl?: number
  handleError?: ApiErrorCallback
  handleSuccess?: ApiSuccessCallback
  immediateTrigger?: boolean
}

interface ApiCaller<T> {
  call: (args?: T) => void
}

interface PaginationArgs {
  pageSize: number
  pageCountCacheTtl: number
  pageCountLastUpdated: number
}

interface MetaArgs<T, TParams = unknown> {
  handleError?: (err: Error) => void
  handleSuccess?: (rs: T) => void
  params?: (QueryParams & TParams) | undefined
  preventFirstCall?: boolean
  triggeringKeys?: string[]
}

interface InternalMetaArgs<T, TParams = unknown> extends MetaArgs<T, TParams> {
  trigger: boolean
  resetTrigger: () => void
  handleError: (err: Error) => void
  handleSuccess: (rs: T) => void
  params: (QueryParams & TParams) | undefined
}
interface ApiCallArgs<T, TArgs> {
  requestDetails?: TArgs
  meta?: MetaArgs<T>
}

type WithPagination<T> = T & {
  pageNumber: number
  pageSize: number
}
