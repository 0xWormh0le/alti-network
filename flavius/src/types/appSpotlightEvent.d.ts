// Display interfaces for each kind of event types are represented below

// event type: people
interface DisplayAppSpotlightPeople {
  avatar: string
  name: string
  email: string
}

// event type: downloads
interface DisplayAppSpotlightEventDownloads {
  view: IFile
  file: IFile
  owner: IPerson
  datetime: number
}

interface BaseAppSpotlightEvent {
  targetPeople: IPerson[]
  files: IFile[]
  eventName: string
  eventDescription: string
  severity: SeverityRange
  datetime: number
}

interface AppSpotlightEvent extends BaseAppSpotlightEvent {
  eventId: string
}

// event type: risks
interface DisplayAppSpotlightRisk extends BaseRisk {
  view: FileResponseWithRisk
  file: FileCellWithRisk
  description: DisplayRiskDescription
  creator: IPerson
  owner: FileOwner
  action: RiskAction
}
