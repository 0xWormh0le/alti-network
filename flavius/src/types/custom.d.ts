declare module 'react-chartist'
declare module 'react-router-modal'
declare module 'react-bookmark'
declare module 'react-google-login'
declare module 'aws-amplify-react'
declare module 'babel-plugin-require-context-hook/register'

type RiskThresholdActorType = 'internal' | 'external'

type Exposure = 'internal' | 'external' | 'unknown' | 'none'

type LinkVisibility = Exposure | 'user' | 'group' | 'internal_discoverable' | 'external_discoverable' | 'sensitive'

type PasswordScore = 0 | 1 | 2 | 3 | 4

type WrapperTypeOptions = 'page' | 'modal'

type GoogleAuthResponse = GoogleLoginResponse | GoogleLoginResponseOffline

type SubNavType =
  | 'allActivity'
  | 'atRiskFilesOwned'
  | 'collaborators'
  | 'appDownloads'
  | 'personDownloads'
  | 'filesAccessible'
  | 'filesSharedBy'
  | 'filesSharedWith'
  | 'risks'
  | 'risksCreated'

type AppSpotlightSubNavType = 'authorizedBy' | 'fileDownloads' | 'associatedRisks'

interface PersonStatistics {
  labels: number[]
  series: {
    [key in SubNavType]: number[]
  }
  stats: {
    [key in SubNavType]: number
  }
}

interface SvgProps {
  className: string
  alt?: string
  text?: string
  onClick?: (...args: any[]) => void
}

interface UserAttributes {
  email: string
  email_verified: boolean
  name: string
  phoneNumber: string
  phoneNumber_verified: boolean
  sub: string
  'custom:is_trial'?: boolean
}

interface AuthenticatedUser {
  attributes: UserAttributes
  id: string
  username: string
}

interface GoogleAuthenticatedUser {
  email: string
  id: string
  name: string
}

interface SettingsUser {
  attributes: UserAttributes
  username: string
  preferredMFA: string
}

interface Name {
  givenName: string
  familyName: string
  fullName: string
}
interface Avatar {
  url: string
  url_etag: string
}

interface Phone {
  number: string
  kind: UsageKind
  primary: boolean
}

interface Notes {
  content: string
  content_type: string
}

interface Email {
  address?: string
  kind: UsageKind
  primary: boolean
  accessCount: number
  riskCount: number
  lastDeletedPermissions: number
}

interface UserId {
  platform: string
  value: string
}
interface IPerson {
  accessLevel: AccessLevel
  name?: Name
  primaryEmail: Email
  projectId: string
  userKind: UserKind
  altnetId?: string
  externalCount?: number
  recoveryEmail?: Email
  phones?: Phone[]
  externalIds?: UserId[]
  avatar?: Avatar
  accessCount?: number
  riskCount?: number
  internal?: boolean // deprecated thanks to userKind
  anonymous?: boolean
  internalCount?: number
  emails?: Email[]
  orgUnitPath?: string
  isEnrolledInMFA?: boolean
  etag?: string
  creationTime?: number
  lastLoginTime?: number
  lastModifiedTime?: number
  notes?: Notes
  lastRemovedPermissions?: number | null
  title?: string
  department?: string
  phoneNumber?: string
}

interface DisplayRiskDescription {
  text: string
  email?: string
  riskId: string
  riskTypeId: RiskTypeId
  pluginId: string
  pluginName: string
  personId?: string
  platformId: string
  incidentDate?: Maybe<number>
}

interface ISensitive {
  fileCount: number
  riskId: string
  riskTypeId: number
}

interface SensitiveMultiFile extends ISensitive {
  sensitivePhrases?: SensitivePhraseMultiFile
}

interface SensitiveSingleFile extends ISensitive {
  sensitivePhrases?: SensitivePhraseSingleFile
}

interface SensitivePhraseMultiFile {
  ccNumFileCount: number
  sensitiveKeywordsFileCount: number
  ssnFileCount: number
  sensitiveKeywords: undefined
}

interface SensitivePhraseSingleFile {
  ccNum: number
  sensitiveKeywordsFileCount: undefined
  sensitiveKeywords: SensitiveObjectKeywordDetail[] | IValue<{ [k: number]: SensitiveObjectKeywordDetail }>
  ssn: number
}

interface IPhrase {
  count: number
  keyword: string
}

interface SensitiveObjectKeywordDetail {
  keyword: string
  count: number
}

interface SensitivePhrase {
  id?: string
  phrase: string
  exact: boolean
}

type SensitivePhrasesResponse = ServerResponseData<{
  sensitivePhrases: SensitivePhrase[]
}>

interface RiskInsight {
  file?: IFile
  person?: IPerson
  count: number
}

interface Application {
  name: string
  id: string
  marketplaceURI: string
  imageURI: string
  grants: string[]
  appId?: string
}

interface ResolutionStatus {
  active: number
  completed: number
  failed: number
  pending: number
  totalCount: number
}

type IMap<T> = {
  get: (x: keyof T) => any
}

interface IValue<T> {
  get: (x: keyof T) => any
}

interface BasicPlatformData {
  platformId: string
  platformName: string
  canConnect: boolean
  resourcesLink?: string
}

interface CompanyInfo {
  domains: string[]
  name: string
}

type Maybe<T> = T | null

type PaginationMeta = {
  pageCount: number
  pageCountCacheTTL: number
  pageCountLastUpdated: number
  pageNumber: number
  pageSize: number
}

type ServerResponseData<T> = {
  orderBy: SortingOption
  sort: SortOrder
} & PaginationMeta &
  T

type Dictionary<T> = { [k: string]: T }

type Primitive = string | number | boolean | bigint

type ValidateFn = (value: string) => string | null
