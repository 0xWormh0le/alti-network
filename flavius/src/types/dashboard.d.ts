interface DashviewStats {
  [k: string]: {
    [k: string]: string | number | null
  }
}
