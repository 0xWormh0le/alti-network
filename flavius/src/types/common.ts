export enum UsageKind {
  personal = 'personal',
  work = 'work'
}

export enum UserKind {
  user = 'user',
  person = 'person',
  serviceAccount = 'serviceAccount'
}

export enum AccessLevel {
  member = 'member',
  manager = 'manager',
  owner = 'owner',
  admin = 'admin'
}

export enum RiskCategoryType {
  SHARING_RISKS = 'Sharings',
  RELATIONSHIP_RISKS = 'Relationships',
  ACTIVITY_RISKS = 'Actions',
  INFORMATIONAL_RISKS = 'Informatives'
}

export enum FileActivityType {
  ACCESS_GRANTED = 'ACCESS_GRANTED',
  CHANGE_FOLDER_PERMISSION = 'CHANGE_FOLDER_PERMISSION',
  CONTENT_ACCESS = 'CONTENT_ACCESS',
  DOWNLOAD = 'DOWNLOAD',
  FILE_WATERMARKED_DOWNLOAD = 'FILE_WATERMARKED_DOWNLOAD',
  ITEM_DOWNLOAD = 'ITEM_DOWNLOAD',
  ITEM_PREVIEW = 'ITEM_PREVIEW',
  ITEM_SHARED = 'ITEM_SHARED',
  ITEM_SHARED_CREATE = 'ITEM_SHARED_CREATE',
  ITEM_SHARED_UPDATE = 'ITEM_SHARED_UPDATE',
  ITEM_SYNC = 'ITEM_SYNC',
  SHARE = 'SHARE'
}

export enum PermissionRole {
  own = 'own',
  read = 'read',
  write = 'write'
}

export enum PermissionShared {
  external = 'external',
  internal = 'internal',
  none = 'none'
}

export enum PermissionType {
  anyone = 'anyone',
  domain = 'domain',
  user = 'user'
}

export enum PermissionItemStatus {
  active = 'ACTIVE',
  pending = 'PENDING',
  removed = 'REMOVED',
  cannotBeRemoved = 'CANNOT_BE_REMOVED'
}

export enum SensitiveContentStatus {
  DETECTION_PENDING = 'pending',
  FOUND_SENSITIVE_CONTENT = 'yes',
  FOUND_NO_SENSITIVE_CONTENT = 'no',
  FAILED_DETECTION = 'failed'
}

export enum FileGridNavType {
  TIMELINE = 'timeline',
  INTERNAL = 'internal',
  EXTERNAL = 'external',
  SENSITIVE = 'sensitive',
  FILES_IN_FOLDER = 'filesInFolder',
  DETAILS = 'details'
}
