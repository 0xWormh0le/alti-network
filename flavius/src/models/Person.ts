import { UserKind, AccessLevel } from 'types/common'
import UI_STRINGS from 'util/ui-strings'
import anonymousUserIcon from 'icons/anonymous-user.svg'
class Person implements IPerson {
  public accessLevel: AccessLevel
  public name: Name
  public projectId: string
  public primaryEmail: Email
  public userKind: UserKind
  public altnetId?: string
  public recoveryEmail?: Email
  public phones: Phone[]
  public externalIds: UserId[]
  public avatar?: Avatar
  public internal?: boolean // deprecated thanks to userKind
  public anonymous?: boolean
  public internalCount?: number
  public externalCount?: number
  public emails: Email[]
  public accessCount?: number
  public riskCount?: number
  public orgUnitPath?: string
  public isEnrolledInMFA?: boolean
  public etag?: string
  public creationTime?: number
  public lastLoginTime?: number
  public lastModifiedTime?: number
  public notes?: Notes
  public lastRemovedPermissions?: number | null
  public title?: string
  public department?: string
  public phoneNumber?: string

  constructor(person: IPerson) {
    this.altnetId = person.altnetId
    this.projectId = person.projectId
    this.userKind = person.userKind
    this.accessLevel = person.accessLevel
    this.name = person.name || {
      givenName: '',
      familyName: '',
      fullName: ''
    }
    this.primaryEmail = person.primaryEmail
    this.recoveryEmail = person.recoveryEmail
    this.phones = person.phones || []
    this.externalIds = person.externalIds || []
    this.orgUnitPath = person.orgUnitPath
    this.etag = person.etag
    this.isEnrolledInMFA = person.isEnrolledInMFA
    this.creationTime = person.creationTime
    this.lastLoginTime = person.lastLoginTime
    this.lastModifiedTime = person.lastModifiedTime
    this.notes = person.notes
    this.avatar =
      this.name.fullName === UI_STRINGS.RISKS.ANONYMOUS_USER ? { url: anonymousUserIcon, url_etag: '' } : person.avatar
    this.title = person.title
    this.department = person.department
    this.phoneNumber = person.phoneNumber
    this.internal = person.internal
    this.anonymous = !person.primaryEmail?.address || person.primaryEmail?.address?.indexOf('@') === -1 // altnetId: person.personId?.indexOf('@') === -1 &&
    this.internalCount = person.internalCount
    this.externalCount = person.externalCount
    this.emails = person.emails || []
    this.accessCount = person.accessCount
    this.riskCount = person.riskCount
    this.lastRemovedPermissions = person.lastRemovedPermissions
  }

  public get displayName(): string {
    if (this.name.givenName && this.name.familyName) {
      return `${this.name.givenName} ${this.name.familyName}`
    } else if (this.anonymous) {
      return UI_STRINGS.RISKS.ANONYMOUS_USER
    } else if (this.primaryEmail?.address) {
      const emailFirst = this.primaryEmail?.address.split('@')
      return emailFirst && emailFirst.length > 0 ? emailFirst[0] : this.primaryEmail?.address
    } else {
      return this.primaryEmail?.address ?? '' // altnetId:
    }
  }

  /*
   * For transform value from Map to Object in griddle-react custom components, e.g. SimplePersonCell, ActorCell
   * This is a workaround solution, once we refactor the table to NOT use griddle-react, this method shall be gone
   */
  static fromMap(personMap: { get: (x: keyof IPerson) => any }): Person {
    return new Person({
      primaryEmail: personMap.get('primaryEmail') && (Object.fromEntries(personMap.get('primaryEmail')) as Email),
      recoveryEmail: personMap.get('recoveryEmail') && (Object.fromEntries(personMap.get('recoveryEmail')) as Email),
      avatar: personMap.get('avatar') && (Object.fromEntries(personMap.get('avatar')) as Avatar),
      name: personMap.get('name') && (Object.fromEntries(personMap.get('name')) as Name),
      internal: personMap.get('internal'),
      internalCount: personMap.get('internalCount'),
      externalCount: personMap.get('externalCount'),
      emails:
        personMap.get('emails') &&
        (Array.from(personMap.get('emails')).map((email: any) => Object.fromEntries(email)) as Email[]),
      accessCount: personMap.get('accessCount'),
      riskCount: personMap.get('riskCount'),
      altnetId: personMap.get('altnetId'),
      projectId: personMap.get('projectId'),
      accessLevel: personMap.get('accessLevel'),
      userKind: personMap.get('userKind'),
      phones:
        personMap.get('phones') &&
        (Array.from(personMap.get('phones')).map((email: any) => Object.fromEntries(email)) as Phone[]),
      externalIds:
        personMap.get('externalIds') &&
        (Array.from(personMap.get('externalIds')).map((email: any) => Object.fromEntries(email)) as UserId[]),
      orgUnitPath: personMap.get('orgUnitPath'),
      etag: personMap.get('etag'),
      isEnrolledInMFA: personMap.get('isEnrolledInMFA'),
      creationTime: personMap.get('creationTime'),
      lastLoginTime: personMap.get('lastLoginTime'),
      lastModifiedTime: personMap.get('lastModifiedTime'),
      notes: personMap.get('notes') && (Object.fromEntries(personMap.get('notes')) as Notes)
    })
  }
}

export default Person
