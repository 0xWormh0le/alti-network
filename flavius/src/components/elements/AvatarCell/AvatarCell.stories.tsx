import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import AvatarCell, { AvatarCellProps } from './AvatarCell'
import { UsageKind } from 'types/common'
import Person from 'models/Person'

export default {
  title: 'Elements/AvatarCell',
  component: AvatarCell
} as Meta

const Template: Story<AvatarCellProps> = (args) => <AvatarCell {...args} />

export const Default = Template.bind({})
Default.args = {
  person: {
    name: {
      givenName: 'Bobbie',
      familyName: 'Hawaii',
      fullName: 'Bobbie Hawaii'
    },
    displayName: 'Bobbie Hawaii',
    primaryEmail: {
      address: 'bobbie@thoughtlabs.io',
      kind: UsageKind.personal,
      primary: true,
      accessCount: 0,
      riskCount: 0,
      lastDeletedPermissions: 0
    },
    recoveryEmail: {
      address: 'test@email.com',
      kind: UsageKind.personal,
      primary: true,
      accessCount: 0,
      riskCount: 0,
      lastDeletedPermissions: 0
    },
    riskCount: 2,
    internal: false
  } as Person
}
