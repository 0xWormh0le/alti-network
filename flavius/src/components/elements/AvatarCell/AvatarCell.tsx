import React from 'react'
import Person from 'models/Person'
import { BaseAvatar } from '@altitudenetworks/component-library/'
import CONSTANTS from 'util/constants'

export interface AvatarCellProps {
  person: Person
}

const AvatarCell: React.FC<AvatarCellProps> = ({ person }) => (
  <div className='AvatarCell'>
    <BaseAvatar src={person.avatar} name={person.displayName} colorList={CONSTANTS.COLOR_LIST} />
  </div>
)

export default AvatarCell
