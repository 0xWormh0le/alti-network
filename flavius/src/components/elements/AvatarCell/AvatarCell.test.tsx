import React from 'react'
import { render } from '@testing-library/react'
import AvatarCell, { AvatarCellProps } from './AvatarCell'
import { UsageKind } from 'types/common'
import Person from 'models/Person'

describe('AvatarCell', () => {
  it('renders correctly', () => {
    const props: AvatarCellProps = {
      person: {
        name: {
          givenName: 'Bobbie',
          familyName: 'Hawaii',
          fullName: 'Bobbie Hawaii'
        },
        displayName: 'Bobbie Hawaii',
        primaryEmail: {
          address: 'bobbie@thoughtlabs.io',
          kind: UsageKind.personal,
          primary: true,
          accessCount: 0,
          riskCount: 0,
          lastDeletedPermissions: 0
        },
        recoveryEmail: {
          address: 'test@email.com',
          kind: UsageKind.personal,
          primary: true,
          accessCount: 0,
          riskCount: 0,
          lastDeletedPermissions: 0
        },
        riskCount: 2,
        internal: false
      } as Person
    }
    const { container } = render(<AvatarCell {...props} />)
    expect(container).toMatchSnapshot()
  })
})
