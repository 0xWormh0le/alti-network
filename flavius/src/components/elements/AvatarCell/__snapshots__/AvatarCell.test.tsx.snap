// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarCell renders correctly 1`] = `
<div>
  <div
    class="AvatarCell"
  >
    <div
      class="BaseAvatar "
    >
      <div
        class="BaseAvatar__text"
        style="background-color: rgb(0, 131, 45);"
      >
        <div>
          B
          H
        </div>
      </div>
    </div>
  </div>
</div>
`;
